### PR TITLE
expose debug executor getter by version

### PR DESCRIPTION
**Pull request by oxade**
## Description 

In some cases we want to invoke specific versions of the executor in debug mode regardless of the protocol version.

## Test Plan 

Unit

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: sui-execution/src/lib.rs [hunk 1]
```diff
@@ -15,20 +15,37 @@ pub mod verifier;
 mod latest;
 mod v0;
 
+pub const MIN_EXECUTOR_VERSION: u64 = 0;
+pub const LATEST_EXECUTOR_VERSION: u64 = 1;
```
Comment (amnn) [inline sui-execution/src/lib.rs:19]: Is this part of the PR necessary?  It complicates automating the creation of new versions of the executor interface.

File: sui-execution/src/lib.rs [hunk 2]
```diff
@@ -43,10 +60,31 @@ pub fn verifier<'m>(
...
         0 => Box::new(v0::Verifier::new(protocol_config, is_metered, metrics)),
-        1 => Box::new(latest::Verifier::new(protocol_config, is_metered, metrics)),
+        LATEST_EXECUTOR_VERSION => {
+            Box::new(latest::Verifier::new(protocol_config, is_metered, metrics))
+        }
         v => panic!("Unsupported execution version {v}"),
     }
 }
+
+/// Sometimes we want to invoke specific versions of the executor regardless of the protocol version
```
Comment (amnn) [inline sui-execution/src/lib.rs:75]: Could you share a bit more of when we want to change the executor version but not the rest of the protocol config?  Generally the execution version could be coupled with other aspects of the protocol, and I would expect it to be overridden by passing in a protocol config for a different version, or by setting a protocol config override, but maybe I'm missing a use case here.

If we can re-use the test seams that `ProtocolConfig` already provides, that would be preferable.

Comment (oxade) [reply sui-execution/src/lib.rs:75]: You are correct that we can use protocol config overrides.
I was not aware of that feature.

**Comment by oxade** [comment]
Can be done by overriding specific field in protocol config.

File: sui-execution/src/lib.rs (hunk 2 above)
Comment (oxade) [reply sui-execution/src/lib.rs:75]: A concrete example is: in many cases people make code changes on latest but want to see the impact of their changes on a historical transaction in replay.

**Pull request closed without merging**